docs/lesson-plans/week-7.md

```diff
@@ -0,0 +1,692 @@
+  
+## Databases! 
+
+Databases are persistent storage. That means that they continue to hold application data on disk, even if your app stops
+running (whether intentionally or unintentionally). They have APIs, mostly through special software Open DataBase 
+Connectivity (ODBC) and Java DataBase Connectivity (JDBC, what we'll be using). The type of database we're going to be 
+working with is called a relational database, which means it acts like a collection of Excel sheets. It also has a nice
+little syntax for performing operations called SQL (pronounced "see-quel" or "ess-que-ell"). There are also 
+other types of DBs, called NoSQL databases, which act a little differently, but we'll just stick with relational in this
+program.
+
+### Excel Sheets to Database Tables
+
+When you think of Excel, you think of a bunch of rows/entries and columns/fields which make up a sheet. In terms of DBs, 
+each table is like a separate sheet containing pretty uniform data. Each column (the up/down ones) acts like a separate 
+field. You specify a datatype and name for your columns, and then as data gets entered, rows (the long left/right) ones 
+get created, read, updated, and deleted (remember CRUD). One thing to note is that in relational databases, unlike 
+Excel, you don't have an unlimited amount of columns that can be automatically added. When you create your table, it has 
+a set structure, meaning you can only add/remove/update columns by changing the structure of the table. You can, 
+however, add an unlimited number of rows in theory, though you'll run out of space and your queries will start to slow 
+down after a while. 
+
+Here's a reminder of what an Excel sheet can look like:
+
+| These | are | your | columns | :) | 
+| :---: | :---: | :---: | :---: | :---: |
+| T | h | e | s | e |
+| a | r | e | :) | :) |
+| y | o | u | r | :) |
+| r | o | w | s | :) |
+
+### Data Types in SQL
+
+In SQL, when making new columns on a table, altering columns, and dealing with table data, you need to be aware of the
+data types of the variables you're working with. It's somewhat similar to working with types in Java, however, you don't
+need to declare types all over the place. It matters most when you're trying to perform operations on types and getting 
+the data using the JDBC/ODBC. For example, if you try to perform a query where you're adding two strings, you'll receive 
+an error since strings can't be added. You can concatenate two strings, but addition is for number types. 
+
+There are many types available, especially in PostgreSQL, but below are a few of the most commonly used ones.
+
+#### SERIAL
+
+The `SERIAL` type is an `INTEGER` under the hood, usually used as an ID. It's a 
+PostgreSQL specific data type, which is usually just an `INTEGER` with the `AUTO INCREMENT` modifier in most other 
+databases. This is really useful for integer-typed
+`PRIMARY KEYS` in other databases, since it automatically creates a new unique ID for you.
```

Comment:
It might be nice to mention how we will go over these later! We don't need people to search what 'AUTO INCREMENT' or 'PRIMARY KEYS' are by themselves when they get to this point, since we explain everything later in this doc!

Reply:
i second this^